numpy/random/mtrand/mtrand.pyx

```diff
@@ -964,20 +964,26 @@ cdef class RandomState:
             high = low
             low = 0
 
+        # '_randint_type' is defined in
+        # 'generate_randint_helpers.py'
         key = np.dtype(dtype).name
-        if not key in _randint_type:
+        if key not in _randint_type:
             raise TypeError('Unsupported dtype "%s" for randint' % key)
+
         lowbnd, highbnd, randfunc = _randint_type[key]
 
-        if low < lowbnd:
+        ilow = int(low)
+        ihigh = int(high)
+
+        if ilow < lowbnd:
```

Comment:
Indeed, a lot of "quirks" (i.e. bugs) arose when I started working on this PR in general.